packages/react/src/components/Authenticator/FederatedSignIn/FederatedSignInButtons/__tests__/FederatedSignInButton.test.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import { fireEvent, render, screen } from '@testing-library/react';
+
+import { FederatedIdentityProviders } from '@aws-amplify/ui';
+import * as ReactCoreModule from '@aws-amplify/ui-react-core';
+
+import { FederatedSignInButton } from '../FederatedSignInButton';
+
+const SUPPORTED_PROVIDERS = [
+  FederatedIdentityProviders.Amazon,
+  FederatedIdentityProviders.Apple,
+  FederatedIdentityProviders.Facebook,
+  FederatedIdentityProviders.Google,
+];
+
+const useAuthenticatorMock = {
+  toFederatedSignIn: jest.fn(),
```

Comment:
To simplify the tests, consider just assigning `toFederatedSignIn` outside of `useAuthenticatorMock`, example:
```
const toFederatedSignIn = jest.fn();

jest
  .spyOn(ReactCoreModule, 'useAuthenticator')
  .mockReturnValue({ toFederatedSignIn });
```

That prevents the need for destructuring `toFederatedSignIn` on line 53

Reply:
That's a nice idea!

Reply:
Done, [863961a](https://github.com/aws-amplify/amplify-ui/pull/2706/commits/863961adb979ad72786d9195d639299b12ad86ab)